### ISSUE TITLE
(MTK) RIL crash on some Devices

### ISSUE THREAD

**Issue by Maanush2004** [phhusson/treble_experimentations#530]
I saw that some mtk device's ril is getting crashed. My device is one of it too. It got fixed once I changed the class to main from core and removed oneshot in rild.rc. From then I didn't face any RIL crash. Not sure which fixed the RIL crash I don't think it is the class I changed. So implement what feels right to you:relaxed:

**Comment by zainarbani** [comment]
@Maanush2004 rild.rc or mtkfusionrild.rc (oreo) ?

**Comment by Maanush2004** [comment]
@zainarbani rild.rc

**Comment by Maanush2004** [comment]
I think mtkfusionrild also uses rild.rc

**Comment by zainarbani** [comment]
not sure, but only found mtkfusionrild on my redmi 6a (oreo)

**Comment by Maanush2004** [comment]
I'm having that dump. The bin is mtkfusionrild but the rc file for it is rild.rc

**Comment by zainarbani** [comment]
yeah you're right

**Comment by phhusson** [comment]
Didn't you say that you did that because you found it in some vendors?

My pie mediatek device has `oneshot`

**Comment by Maanush2004** [comment]
I don't really know if it works for all it did for me and mine is mtkrild not mtkfusionrild

**Comment by phhusson** [comment]
That's not my question.
My question is where does your inspiration come from to try removing `oneshot`?

**Comment by Maanush2004** [comment]
I checked the rc file of rild in aosp source and there was no oneshot. If I'm not wrong oneshot means it only tries to start the service once right?

**Comment by phhusson** [comment]
Yeah. Which means that without `oneshot`, whenever rild crashes (and only in that case, not for other error cases), it will restart rild.
So that means that, if this fix works, at the very least, you lost your connection/calls/whatever for like thirty seconds.

I'm not against implementing this workaround (I still need to figure out how to do that, but that's something else), but that's really just a workaround, not a fix, since it isn't used by mediatek.

**Comment by Maanush2004** [comment]
Yeah that is why I asked about it cause I loose signals for 10-20 secs sometimes and get them back and yes it is not a fix but a workaround

**Comment by phhusson** [comment]
You mean you edited mtkfusionrild.rc and removed oneshot line?

**Comment by Maanush2004** [comment]
yes that is what I did, but that is not necessary anymore. RIL starting working fine without crashing in your newer releases. I don't remember which release it was but it did get fixed. Moreover removing oneshot I think is more like a workaround by starting the RIL service again.

**Comment by phhusson** [comment]
I'm surprised, I don't think I've done anything to fix this issue, that I myself witnessed. (Not often because you really need to daily drive a phone to see it...).

Restarting RIL is a fine fix by me.

I'll leave this open for the moment
